### PR TITLE
feat: better handling non admin layout setting

### PR DESCRIPTION
**Pull request by koshuang**
#61

### CODE REVIEW

File: examples/next-starter/src/app/configs/appConfig.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
+import { AppConfig } from '@onr/core';
 import getConfig from 'next/config';
+
 const { publicRuntimeConfig } = getConfig() as {
   publicRuntimeConfig: Record<'processEnv' | string, unknown>;
 };
 
-export const appConfig = {
+export const appConfig: AppConfig = {
```
Comment (TianyiLi) [inline examples/next-starter/src/app/configs/appConfig.ts:8]: if the `appConfig` is constant and read-only value, I suggest using `as const` make the object cannot be modified.

**Review by TianyiLi — APPROVED** [review]
r+

File: examples/next-starter/src/app/configs/appConfig.ts [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
   apiBaseUrl: `${publicRuntimeConfig.processEnv.NEXT_PUBLIC_API_URL}/api`,
+
   apiKey: publicRuntimeConfig.processEnv.NEXT_PUBLIC_API_KEY || '',
+
+  // prettier-ignore
+  'full-page-routes': [
```
Comment (l443018) [inline examples/next-starter/src/app/configs/appConfig.ts:14]: I feel "full-page-routes"  is a little ambiguous, how about add comment to describe it? like "routes not show on dashboard navbar"

**Comment by l443018** [comment]
r+ with comment thanks!

File: examples/next-starter/src/app/configs/appConfig.ts (hunk 1 above)
Comment (koshuang) [reply examples/next-starter/src/app/configs/appConfig.ts:8]: Good point, thanks!

**Pull request merged**